redhat-appstudio/service-provider-integration-operator
# Add parameters validation on token upload endpoint

### What does this PR do?
Add namespace and token name parameters validation on token upload endpoint.
- Namespaces must comply with DNS1123 labels requirement. (see https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#namespaces-and-dns)
- Tokens must comply general label values requirement, since the SPI access token names are used as a
label values in the linked objects.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-364

### How to test this PR?


```
quay.io/redhat-appstudio/pull-request-builds:spi-operator-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
Perform a token upload POST with invalid parameters.  Expect `400` response with the explanation.
